aea/cli/generate_all_protocols.py

```diff
@@ -384,6 +401,24 @@ def _process_packages_protocol(
     _generate_protocol(package_path)
     _fix_generated_protocol(package_path)
 
+    packages_dir_parent = package_path.parent.parent.parent.parent
+    if root_dir != packages_dir_parent:
+        if root_dir not in packages_dir_parent.parents:
+            raise ValueError(
+                "Packages dir should be a sub directory of the root directory."
+            )
+
+        relative_import_path = ".".join(
+            packages_dir_parent.parts[len(root_dir.parts) :]
+        )
+        replacements = [
+            (
+                f"from packages.fetchai.protocols.{package_path.name}",
+                f"from {relative_import_path}.packages.fetchai.protocols.{package_path.name}",
+            )
+        ]
+        replace_in_directory(package_path.name, replacements)
```

Comment:
This will replace the autogenerated import statements with module path relative to the root dir. In this case 
`packages.fetchai.protocols.t_protocol` will become `test.data.packages.fetchai.protocols.t_protocol`